spicy/include/rt/parsed-unit.h

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2020 by the Zeek Project. See LICENSE for details.
+
+#pragma once
+
+#include <hilti/rt/type-info.h>
+#include <hilti/rt/types/reference.h>
+
+namespace spicy::rt {
+
+/**
+ * Type-erased wrapper around parsed unit instances.
+ *
+ * Initially, this will be unbound, i.e., not refer to any particular
+ * instance. `init()` then binds it to an instance and have
+ * `ParsedUnit` hold a strong reference to it.
+ */
+class ParsedUnit : public hilti::rt::type_info::value::Parent {
+public:
+    /** Returns typed access to the contained instance. */
+    template<typename T>
+    const T& get() const {
+        if ( auto p = _unit.as<T>() )
+            return *p;
+        else
+            throw hilti::rt::NullReference("parsed unit not set");
+    }
+
+    /**
+     * Returns the instance and its type in a value representation suitable
+     * to use with the `type-info` API for iteration over the fields.
+     */
+    hilti::rt::type_info::Value value() const { return {_ptr, _ti, *this}; }
```

Comment:
Does it make sense to create a `Value` if we are not initialized and `_ptr` and `_ti` are nil? Maybe we should assert here or emit an internal error.